### DISCUSSION TITLE
how to use `trust proxy` with nextjs api route

### DISCUSSION THREAD

**Discussion by eretica** [vercel/next.js#14761] [Help] [answered]
I'm using the [Next Api routes](https://nextjs.org/docs/api-routes/introduction) with [express-session](https://github.com/expressjs/session#cookiesecure)

How to configure a `trust proxy` to take advantage of the `secure: true` option?

```
  cookie: {
    httpOnly: true,
    sameSite: true,
    secure: true,
  },
```

**Comment by eretica** [comment, marked as answer]
sorry 😭 i resolved!

add `proxy: true`

```
{
  saveUninitialized: true,
  secret: 'keyboard cat',
  resave: false,
  store: new RedisStore({
    client: redisClient,
  }),
  cookie: {
    maxAge: 1000 * 60 * 60 * 24 * 365,
    httpOnly: true,
    sameSite: true,
    secure: process.env.NODE_ENV === 'production',
  },
  proxy: true,  // ← add!
}
```

**Reply by popuguytheparrot** [reply]
Hi. Can you show full example with code?

**Reply by eretica** [reply]
I resolved this discuttion
By sniped last line  `proxy: true`

**Reply by popuguytheparrot** [reply]
I understand it, but i dont understand how to write handler for it in api folder

**Reply by eretica** [reply]
https://qiita.com/eretica/items/16c858d4b48013f560ca
I was write blog. Please show this😭 but japanese language

**Reply by popuguytheparrot** [reply]
NP. Thanks you for sharing article 👍
